editor/plugins/style_box_editor_plugin.cpp

```diff
@@ -31,116 +31,105 @@
-		preview_rect.size -= grid_texture_disabled->get_size() * 2;
+		float grid_button_width = get_theme_icon(SNAME("StyleBoxGrid"), SNAME("EditorIcons"))->get_size().x;
+		Rect2 preview_rect = get_rect();
+		preview_rect = preview_rect.grow(-grid_button_width);
 
-		// Re-adjust preview panel to fit all drawn content
-		Rect2 draw_rect = stylebox->get_draw_rect(preview_rect);
-		preview_rect.size -= draw_rect.size - preview_rect.size;
-		preview_rect.position -= draw_rect.position - preview_rect.position;
+		// Re-adjust preview panel to fit all drawn content.
+		Rect2 drawing_rect = stylebox->get_draw_rect(preview_rect);
+		preview_rect.size -= drawing_rect.size - preview_rect.size;
+		preview_rect.position -= drawing_rect.position - preview_rect.position;
 
-		preview->draw_style_box(stylebox, preview_rect);
+		draw_style_box(stylebox, preview_rect);
 
 		Ref<StyleBoxTexture> sbt = stylebox;
+		// Draw the "grid". Use white lines, as well as subtle black lines to ensure contrast.
 		if (sbt.is_valid() && grid_preview->is_pressed()) {
-			for (int i = 0; i < 2; i++) {
-				Color c = i == 1 ? Color(1, 1, 1, 0.8) : Color(0, 0, 0, 0.4);
-				int x = draw_rect.position.x + sbt->get_margin(SIDE_LEFT) + (1 - i);
-				preview->draw_line(Point2(x, 0), Point2(x, preview->get_size().height), c);
-				int x2 = draw_rect.position.x + draw_rect.size.width - sbt->get_margin(SIDE_RIGHT) + (1 - i);
-				preview->draw_line(Point2(x2, 0), Point2(x2, preview->get_size().height), c);
-				int y = draw_rect.position.y + sbt->get_margin(SIDE_TOP) + (1 - i);
-				preview->draw_line(Point2(0, y), Point2(preview->get_size().width, y), c);
-				int y2 = draw_rect.position.y + draw_rect.size.height - sbt->get_margin(SIDE_BOTTOM) + (1 - i);
-				preview->draw_line(Point2(0, y2), Point2(preview->get_size().width, y2), c);
-			}
+			const Color dark_color = Color(0, 0, 0, 0.4);
+			const Color bright_color = Color(1, 1, 1, 0.8);
+			int x_left = drawing_rect.position.x + sbt->get_margin(SIDE_LEFT);
+			int x_right = drawing_rect.position.x + drawing_rect.size.width - sbt->get_margin(SIDE_RIGHT);
+			int y_top = drawing_rect.position.y + sbt->get_margin(SIDE_TOP);
+			int y_bottom = drawing_rect.position.y + drawing_rect.size.height - sbt->get_margin(SIDE_BOTTOM);
+
+			draw_line(Point2(x_left + 2, 0), Point2(x_left + 2, get_size().height), dark_color);
+			draw_line(Point2(x_right + 1, 0), Point2(x_right + 1, get_size().height), dark_color);
+			draw_line(Point2(0, y_top + 2), Point2(get_size().width, y_top + 2), dark_color);
+			draw_line(Point2(0, y_bottom + 1), Point2(get_size().width, y_bottom + 1), dark_color);
+
+			draw_line(Point2(x_left + 1, 0), Point2(x_left + 1, get_size().height), bright_color);
+			draw_line(Point2(x_right, 0), Point2(x_right, get_size().height), bright_color);
+			draw_line(Point2(0, y_top + 1), Point2(get_size().width, y_top + 1), bright_color);
+			draw_line(Point2(0, y_bottom), Point2(get_size().width, y_bottom), bright_color);
```

Comment:
I actually meant adding comments around here, to indicate which lines we're drawing. Can be hard to grasp quickly where the horizontal and vertical lines are. But it's not too critical.